packages/rainbowkit/src/components/RainbowKitProvider/RainbowKitProvider.tsx

```diff
@@ -97,34 +101,36 @@ export function RainbowKitProvider({
             <AvatarContext.Provider value={avatarContext}>
               <AppContext.Provider value={appContext}>
                 <ThemeIdContext.Provider value={id}>
-                  {theme ? (
-                    <div {...createThemeRootProps(id)}>
-                      <style
-                        // eslint-disable-next-line react/no-danger
-                        dangerouslySetInnerHTML={{
-                          // Selectors are sanitized to only contain alphanumeric
-                          // and underscore characters. Theme values generated by
-                          // cssStringFromTheme are sanitized, removing
-                          // characters that terminate values / HTML tags.
-                          __html: [
-                            `${selector}{${cssStringFromTheme(
-                              'lightMode' in theme ? theme.lightMode : theme
-                            )}}`,
-
-                            'darkMode' in theme
-                              ? `@media(prefers-color-scheme:dark){${selector}{${cssStringFromTheme(
-                                  theme.darkMode,
-                                  { extends: theme.lightMode }
-                                )}}}`
-                              : null,
-                          ].join(''),
-                        }}
-                      />
-                      {children}
-                    </div>
-                  ) : (
-                    children
-                  )}
+                  <ModalProvider>
```

Comment:
This is nested at this depth because the modal UI needs access to the other context values.